django/db/models/expressions.py

```diff
@@ -257,6 +257,10 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
     def conditional(self):
         return isinstance(self.output_field, fields.BooleanField)
 
+    @property
+    def ordered(self):
+        return isinstance(self, OrderBy)
```

Comment:
We could move this hook to a separate PR, and change current checks in few places e.g. `db/models/sql/compiler.py`.

Reply:
Makes sense, I'll look into creating a separate PR for this.

Reply:
I took a stab at this and it didn't fit that well in my opinion. To not make the code doing the `isinstance(self, OrderBy)` checks even more complex we would need to add this function to the `F`-class and always return `True`. There were also no places where a `isinstance(foo, OrderBy)` could just be replaced by `foo.ordered` without changing surrounding code. I can still open up a PR to demonstrate this if you like but otherwise I think we should just fall back to using `isinstance(self, OrderBy)` directly here as well.

Reply:
Is there a reason this can't be implemented as `BaseExpression.ordered = False` with a `OrderBy.ordered = True` override like other properties are dealt with? (e.g. `filterable`, `window_compatible`).

This seems like unnecessary bi-directional coupling.

Reply:
That’s much better than the `isinstance`-check but it still have the same issue where we would need to set both `BaseExpression.ordered = False` and `F.ordered = False`. If you feel like that’s OK I’ll open up a PR. I guess we could set `Combinable.ordered = False` but that feels a bit out-of-place IMO.

Reply:
Implemented in #13738.